### PR TITLE
Add theme switch

### LINKED ISSUE #77: Explainable Hotel ratings - App doesn't have an option for Dark Mode (optional)

### PR DESCRIPTION
**Pull request by Mathanraj-Sharma**
Fix #77

### CODE REVIEW

File: explaining-ratings/src/app.py [hunk 1]
```diff
@@ -85,52 +94,69 @@ def handle_filter(q: Q, key: str, val: str):
...
-        sidebar_items[9].dropdown.value = q.args.filter_userProvince
-    if q.args.review:
-        q.client.review = q.args.review
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[0].dropdown.value = q.client.review
-        q.page['original'].image = plot_word_cloud(config.dataset[q.client.review], q)
-
-    if q.client.filters:
-        render_diff_word_cloud(q)
+    if q.args.theme_dark is not None and q.args.theme_dark != q.client.theme_dark:
```
Comment (mturoci) [inline explaining-ratings/src/app.py:118]: Nit
```suggestion
    if q.args.theme_dark and q.args.theme_dark != q.client.theme_dark:
```

File: explaining-ratings/src/app.py [hunk 2]
```diff
@@ -85,52 +94,69 @@ def handle_filter(q: Q, key: str, val: str):
...
+        await update_theme(q)
     else:
-        del q.page['diff']
```
Comment (mturoci) [inline explaining-ratings/src/app.py:134]: What's the reason for including the rest of the handlers in an `else` block? Can it be flattened (else removed)?

**Review by mturoci — CHANGES_REQUESTED** [review]
Thanks @Mathanraj-Sharma! A few small comments from me, looks good otherwise.

File: explaining-ratings/src/app.py (hunk 2 above)
Comment (Mathanraj-Sharma) [reply explaining-ratings/src/app.py:134]: if they are not in `else` block when we switch them it resubmits the filters as well. So the rating diff also getting refreshed

File: explaining-ratings/src/app.py (hunk 1 above)
Comment (Mathanraj-Sharma) [reply explaining-ratings/src/app.py]: We need to have `q.args.theme_dark is not None` to make sure `update_theme` is called for each case 
- `q.args.theme_dark = True` to switch to dark
- `q.args.theme_dark = False` to switch to light

**Comment by Mathanraj-Sharma** [comment]
@mturoci I have improved the conditions to switch theme, please check

File: explaining-ratings/src/app.py (hunk 1 above)
Comment (mturoci) [reply explaining-ratings/src/app.py]: Ah it's a boolean, makes sense then.

File: explaining-ratings/src/app.py (hunk 2 above)
Comment (mturoci) [reply explaining-ratings/src/app.py]: But if the same filters are applied again, shouldn't it result in the very same diff? In which case it doesn't matter if they are rerun or not.

Comment (Mathanraj-Sharma) [reply explaining-ratings/src/app.py]: @mturoci no it doesn't it creates different word clouds (as diff) for each time it get triggered with same filters

**Pull request merged**